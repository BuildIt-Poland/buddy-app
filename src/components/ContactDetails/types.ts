import { RouteComponentProps } from 'react-router-dom';

export type ContactDetailsProps = Pick<RouteComponentProps, 'history'>;
