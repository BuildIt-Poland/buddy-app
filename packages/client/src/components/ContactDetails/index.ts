import ContactDetails from './ContactDetails';

export default ContactDetails;
