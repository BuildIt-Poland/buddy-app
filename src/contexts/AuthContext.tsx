import React from 'react';
import { AuthPayload } from 'buddy-app-schema';

export interface State {
  loading: boolean;
  error: any;
  data: AuthPayload;
  isAuthenticated: boolean;
  isAppBootstrapped: boolean;
}

export interface AuthContextData extends State {
  login: (email: string, password: string) => void;
  logout: () => void;
}

export const defaultState: State = {
  loading: false,
  error: null,
  data: {} as AuthPayload,
  isAuthenticated: false,
  isAppBootstrapped: false,
};

const defaultContext: AuthContextData = {
  ...defaultState,
  login: () => null,
  logout: () => null,
};

const AuthContext = React.createContext<AuthContextData>(defaultContext);

export default AuthContext;
