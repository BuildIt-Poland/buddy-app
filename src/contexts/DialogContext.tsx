import React from 'react';

export interface State {
  isOpen: boolean;
  message: string;
  title: string | undefined;
  onConfirm: (() => void) | undefined;
}

export interface DialogContextData extends State {
  showDialog: (message: string, title?: string, onConfirm?: () => void) => void;
  hideDialog: () => void;
}

export const defaultState: State = {
  isOpen: false,
  title: '',
  message: '',
  onConfirm: () => {},
};

const defaultContext: DialogContextData = {
  ...defaultState,
  showDialog: () => null,
  hideDialog: () => null,
};

const DialogContext = React.createContext<DialogContextData>(defaultContext);

export default DialogContext;
