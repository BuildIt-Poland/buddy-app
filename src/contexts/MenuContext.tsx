import React from 'react';

export interface State {
  isOpen: boolean;
}

export interface MenuContextData extends State {
  toggleMenu: () => void;
}

export const defaultState: State = {
  isOpen: false,
};

const defaultContext: MenuContextData = {
  ...defaultState,
  toggleMenu: () => null,
};

const MenuContext = React.createContext<MenuContextData>(defaultContext);

export default MenuContext;
