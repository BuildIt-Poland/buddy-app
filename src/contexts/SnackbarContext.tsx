import React from 'react';

export interface State {
  isOpen: boolean;
  message: string;
}

export interface SnackbarContextData extends State {
  showSnackbar: (message: string) => void;
  hideSnackbar: () => void;
}

export const defaultState: State = {
  isOpen: false,
  message: '',
};

const defaultContext: SnackbarContextData = {
  ...defaultState,
  showSnackbar: () => null,
  hideSnackbar: () => null,
};

const SnackbarContext = React.createContext<SnackbarContextData>(defaultContext);

export default SnackbarContext;
