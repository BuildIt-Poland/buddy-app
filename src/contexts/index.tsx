import React from 'react';

import { apolloClient } from 'utils';
import { ApolloProvider } from '@apollo/react-hooks';
import theme from 'styles/theme';
import ThemeProvider from '@material-ui/styles/ThemeProvider';
import AuthProvider from 'stores/AuthProvider';
import MenuProvider from 'stores/MenuProvider';
import SnackbarProvider from 'stores/SnackbarProvider';

const AppProviders: React.FC = ({ children }) => {
  return (
    <ApolloProvider client={apolloClient}>
      <ThemeProvider theme={theme}>
        <AuthProvider>
          <MenuProvider>
            <SnackbarProvider>{children}</SnackbarProvider>
          </MenuProvider>
        </AuthProvider>
      </ThemeProvider>
    </ApolloProvider>
  );
};

export default AppProviders;
