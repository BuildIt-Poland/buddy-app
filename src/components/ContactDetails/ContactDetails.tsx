import React from 'react';
import { useParams } from 'react-router-dom';
import Typography from '@material-ui/core/Typography';
import { useQuery } from '@apollo/react-hooks';
import {
  NEWBIE_CONTACT_DETAILS,
  BUDDY_CONTACT_DETAILS,
} from 'graphql/contact-details.graphql';
import { ROUTES } from 'shared/routes';
import { QueryBuddyArgs, QueryNewbieArgs, UserRole } from 'buddy-app-schema';
import { useAuth } from 'contexts/AuthContext';
import UserDetails from 'components/UserDetails';
import { BasicDetailsParams, UserBasicDetails } from 'components/UserMenu/types';
import Box from '@material-ui/core/Box';
import PageContainer from 'components/PageContainer';
import Header, { MenuTypes } from 'components/Header';
import { ContactDetailsProps } from './types';

const ContactDetails: React.FC<ContactDetailsProps> = ({ history }) => {
  const { newbieId } = useParams<QueryNewbieArgs>();
  const { buddyId } = useParams<QueryBuddyArgs>();
  const isCurrentUserDetails = !newbieId && !buddyId;
  const [
    {
      data: { role, userId },
    },
  ] = useAuth();

  const queryByRole = isCurrentUserDetails
    ? {
        [UserRole.Newbie]: {
          query: NEWBIE_CONTACT_DETAILS,
          variables: { newbieId: userId },
          userRole: UserRole.Newbie.toLowerCase(),
          onBackClick: () => history.push(ROUTES.NEWBIE_TASKS_LIST),
        },
        [UserRole.Buddy]: {
          query: BUDDY_CONTACT_DETAILS,
          variables: { buddyId: userId },
          userRole: UserRole.Buddy.toLowerCase(),
          onBackClick: () => history.push(ROUTES.BUDDY_SELECT_NEWBIE),
        },
      }
    : {
        [UserRole.Newbie]: {
          query: BUDDY_CONTACT_DETAILS,
          variables: { buddyId },
          userRole: UserRole.Buddy.toLowerCase(),
          onBackClick: () => history.push(ROUTES.NEWBIE_TASKS_LIST),
        },
        [UserRole.Buddy]: {
          query: NEWBIE_CONTACT_DETAILS,
          variables: { newbieId },
          userRole: UserRole.Newbie.toLowerCase(),
          onBackClick: () =>
            history.push(ROUTES.BUDDY_TASKS_LIST.replace(':newbieId', newbieId)),
        },
      };

  const { query, variables, userRole, onBackClick } = queryByRole[role];
  const { data, loading } = useQuery<UserBasicDetails, BasicDetailsParams>(query, {
    variables,
  });
  const userDetails = data && data[userRole as string];

  return (
    <>
      <Header
        type={MenuTypes && MenuTypes.BACK}
        loading={loading}
        onButtonClick={onBackClick}
      />
      <PageContainer backGroundShape>
        <Box>
          <Typography component='h2' variant='h2'>
            Contact Details
          </Typography>
          {userDetails && <UserDetails details={userDetails} />}
        </Box>
      </PageContainer>
    </>
  );
};

export default ContactDetails;
