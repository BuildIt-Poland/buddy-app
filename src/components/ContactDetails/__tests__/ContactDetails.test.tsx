import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import { MockedProvider } from '@apollo/react-testing';
import { act, create } from 'react-test-renderer';
import waitForExpect from 'wait-for-expect';
import ContactDetails from 'components/ContactDetails';
import {
  buddyContactDetails,
  newbieContactDetails,
  mockedBuddyContext,
  mockedNewbieContext,
} from '__mocks__';
import { AuthProvider } from 'contexts/AuthContext';

jest.mock('@material-ui/core/Typography', () => 'Typography');
jest.mock('@material-ui/core/Box', () => 'Box');
jest.mock('components/UserDetails', () => 'UserDetails');
jest.mock('components/PageContainer', () => 'PageContainer');
jest.mock(
  'components/Header',
  () => (require.requireActual('components/Header').default = () => 'Header')
);

describe('Component - ContactDetails', () => {
  const mockHistory: any = {
    push: jest.fn(),
  };

  describe('when logged in as buddy', () => {
    const path = '/buddy/newbies/1234/details';
    const component = create(
      <MockedProvider
        mocks={newbieContactDetails({ newbieId: '1234' })}
        addTypename={false}
        resolvers={{}}>
        <MemoryRouter initialEntries={[path]}>
          <Route path={'/buddy/newbies/:newbieId/details'}>
            <AuthProvider value={mockedBuddyContext()}>
              <ContactDetails history={mockHistory} />
            </AuthProvider>
          </Route>
        </MemoryRouter>
      </MockedProvider>
    );
    it('renders correctly', async () => {
      expect(component).toMatchSnapshot();

      await act(async () => {
        await waitForExpect(() => {
          expect(component).toMatchSnapshot();
        });
      });
    });
  });

  describe('when logged in as newbie', () => {
    const path = '/newbie/buddy/1234/details';
    const component = create(
      <MockedProvider
        mocks={buddyContactDetails({ buddyId: '1234' })}
        addTypename={false}
        resolvers={{}}>
        <MemoryRouter initialEntries={[path]}>
          <Route path={'/newbie/buddy/:buddyId/details'}>
            <AuthProvider value={mockedNewbieContext()}>
              <ContactDetails history={mockHistory} />
            </AuthProvider>
          </Route>
        </MemoryRouter>
      </MockedProvider>
    );
    it('renders correctly', async () => {
      expect(component).toMatchSnapshot();

      await act(async () => {
        await waitForExpect(() => {
          expect(component).toMatchSnapshot();
        });
      });
    });
  });
});
