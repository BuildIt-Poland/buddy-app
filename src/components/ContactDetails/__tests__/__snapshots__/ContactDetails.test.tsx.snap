// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - ContactDetails when logged in as buddy renders correctly 1`] = `
Array [
  <header
    className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionSticky makeStyles-appBar-2 makeStyles-defaultAppBarBackground-4 makeStyles-withoutShadow-3"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-1 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      className="MuiLinearProgress-root MuiLinearProgress-colorPrimary makeStyles-loaderShadow-6 MuiLinearProgress-indeterminate"
      role="progressbar"
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
        style={Object {}}
      />
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
        style={Object {}}
      />
    </div>
    <Box
      className="makeStyles-paperAppBarBackground-5"
    />
  </header>,
  <PageContainer
    backGroundShape={true}
  >
    <Box>
      <Typography
        component="h2"
        variant="h2"
      >
        Contact Details
      </Typography>
    </Box>
  </PageContainer>,
]
`;

exports[`Component - ContactDetails when logged in as buddy renders correctly 2`] = `
Array [
  <header
    className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionSticky makeStyles-appBar-2 makeStyles-defaultAppBarBackground-4 makeStyles-withoutShadow-3"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-1 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      className="MuiLinearProgress-root MuiLinearProgress-colorPrimary makeStyles-loaderShadow-6 makeStyles-hideLoader-7 MuiLinearProgress-indeterminate"
      role="progressbar"
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
        style={Object {}}
      />
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
        style={Object {}}
      />
    </div>
    <Box
      className="makeStyles-paperAppBarBackground-5"
    />
  </header>,
  <PageContainer
    backGroundShape={true}
  >
    <Box>
      <Typography
        component="h2"
        variant="h2"
      >
        Contact Details
      </Typography>
      <UserDetails
        details={
          Object {
            "email": "tom@wipro.com",
            "id": "1234",
            "name": "Tom Hanks",
            "notes": "Nice guy",
            "phoneNumber": "1234567",
            "photo": "some-url",
            "position": "Dev Ops",
            "role": "NEWBIE",
            "startDate": "2010-01-01",
          }
        }
      />
    </Box>
  </PageContainer>,
]
`;

exports[`Component - ContactDetails when logged in as newbie renders correctly 1`] = `
Array [
  <header
    className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionSticky makeStyles-appBar-2 makeStyles-defaultAppBarBackground-4 makeStyles-withoutShadow-3"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-1 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      className="MuiLinearProgress-root MuiLinearProgress-colorPrimary makeStyles-loaderShadow-6 makeStyles-hideLoader-7 MuiLinearProgress-indeterminate"
      role="progressbar"
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
        style={Object {}}
      />
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
        style={Object {}}
      />
    </div>
    <Box
      className="makeStyles-paperAppBarBackground-5"
    />
  </header>,
  <PageContainer
    backGroundShape={true}
  >
    <Box>
      <Typography
        component="h2"
        variant="h2"
      >
        Contact Details
      </Typography>
      <UserDetails
        details={
          Object {
            "email": "tom@wipro.com",
            "id": "1234",
            "name": "Tom Hanks",
            "phoneNumber": "1234567",
            "photo": "some-url",
            "position": "Dev Ops",
            "role": "BUDDY",
            "startDate": "2010-01-01",
          }
        }
      />
    </Box>
  </PageContainer>,
]
`;

exports[`Component - ContactDetails when logged in as newbie renders correctly 2`] = `
Array [
  <header
    className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionSticky makeStyles-appBar-2 makeStyles-defaultAppBarBackground-4 makeStyles-withoutShadow-3"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-1 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      className="MuiLinearProgress-root MuiLinearProgress-colorPrimary makeStyles-loaderShadow-6 makeStyles-hideLoader-7 MuiLinearProgress-indeterminate"
      role="progressbar"
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
        style={Object {}}
      />
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
        style={Object {}}
      />
    </div>
    <Box
      className="makeStyles-paperAppBarBackground-5"
    />
  </header>,
  <PageContainer
    backGroundShape={true}
  >
    <Box>
      <Typography
        component="h2"
        variant="h2"
      >
        Contact Details
      </Typography>
      <UserDetails
        details={
          Object {
            "email": "tom@wipro.com",
            "id": "1234",
            "name": "Tom Hanks",
            "phoneNumber": "1234567",
            "photo": "some-url",
            "position": "Dev Ops",
            "role": "BUDDY",
            "startDate": "2010-01-01",
          }
        }
      />
    </Box>
  </PageContainer>,
]
`;
